rpcs3/Emu/Cell/Modules/sys_net.cpp

```diff
@@ -150,57 +176,115 @@ namespace sys_net
 		return g_tls_net_data.ref(&_tls_data_t::_h_errno);
 	}
 
+	// Error helper functions
+	s32 get_last_error()
+	{
+		// Convert the error code for socket functions to a one for sys_net
+#ifdef _WIN32
+		switch (WSAGetLastError())
+		{
+		case WSAEWOULDBLOCK:
+			return SYS_NET_EWOULDBLOCK;
+
+		default:
+			throw EXCEPTION("Unknown Win32 socket error: %d", WSAGetLastError());
+		}
+#else
+		switch (errno)
+		{
+		case EWOULDBLOCK:
+			return SYS_NET_EWOULDBLOCK;
+
+		default:
+			throw EXCEPTION("Unknown Unix socket error: %d", errno);
+		}
+
+		return errno;
+#endif
+	}
+
 	// Functions
 	s32 accept(s32 s, vm::ptr<sockaddr> addr, vm::ptr<u32> paddrlen)
 	{
 		libnet.warning("accept(s=%d, family=*0x%x, paddrlen=*0x%x)", s, addr, paddrlen);
-		s = g_socketMap[s];
+		s = idm::get<sys_net_socket>(s)->s;
+
+		s32 ret;
 
-		if (!addr) {
-			int ret = ::accept(s, nullptr, nullptr);
-			get_errno() = getLastError();
-			return ret;
+		if (!addr)
+		{
+			ret = ::accept(s, nullptr, nullptr);
+
+			if (ret < 0)
+			{
+				libnet.error("accept(): error %d", get_errno() = get_last_error());
+				return -1;
```

Comment:
Not really, it seems all sys_net functions return CELL_* -alike return codes like SYS_NET_ERROR_ENOENT = 0x80010202 (it's 0x80010200 + SYS_NET_ENOENT).